src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1118,23 +1118,20 @@ internal bool HasCollectionExpressionApplicableAddMethod(SyntaxNode syntax, Type
                                 addMethods = [];
                                 result = false;
                             }
-                            else if (finalApplicableCandidates.Length == 1 &&
-                                     tryEarlyBindSingleCandidateInvocationWithDynamicArgument(addMethodBinder, syntax, expression, methodGroup, diagnostics, resolution, finalApplicableCandidates[0], out var addMethod) is bool earlyBoundResult)
-                            {
-                                addMethods = addMethod is null ? [] : [addMethod];
-                                result = earlyBoundResult;
-                            }
                             else
                             {
-                                Debug.Assert(finalApplicableCandidates.Length > 0);
-
                                 addMethods = filterOutBadGenericMethods(addMethodBinder, syntax, methodGroup, analyzedArguments, resolution, finalApplicableCandidates, ref useSiteInfo);
                                 result = !addMethods.IsEmpty;
 
                                 if (!result)
                                 {
                                     diagnostics.Add(ErrorCode.ERR_CollectionExpressionMissingAdd, syntax, methodGroup.ReceiverOpt.Type);
                                 }
+                                else if (addMethods.Length == 1)
```

Comment:
It's hard to tell whether we're keeping the correspondence with `BindMethodGroupInvocation`. What does this correspond to?

Reply:
> It's hard to tell whether we're keeping the correspondence with ```BindMethodGroupInvocation```. What does this correspond to?

It corresponds to the code at the end of ```bindInvocationExpressionContinued``` below. We are not going after exact match with ```BindMethodGroupInvocation``` because the argument that we are using is fake and we aways end up here we we have a single applicable candidate. Before the change we were ending up in ```bindInvocationExpressionContinued``` through ```tryEarlyBindSingleCandidateInvocationWithDynamicArgument```.